tiledb/sm/query/readers/reader_base.cc

```diff
@@ -1572,10 +1656,48 @@ Status ReaderBase::unfilter_tiles(
           // Unfilter 't' for fixed-sized tiles, otherwise unfilter both 't' and
           // 't_var' for var-sized tiles.
           if (!var_size) {
-            if (!nullable)
-              RETURN_NOT_OK(unfilter_tile(name, t));
-            else
+            if (!nullable) {
+              if (array_schema_.bitsort_filter_attr()) {
```

Comment:
Is this the same code as above? If so we need to combine.

Reply:
Just did!

Reply:
The main issue with putting this in a function is that since the GlobalCmpQB struct will use references to both the domain and the DomainBuffersView objects (and the latter seems to not have a copy constructor), we have to create the DomainBuffersView object in the reader base function, otherwise it will be corrupted data when we get to the bitsort filter. I'm currently trying to find out how to mitigate this right now but help is always nice!